src/ttl_table.c

```diff
@@ -0,0 +1,142 @@
+#include "ttl_table.h"
+#include "rmalloc.h"
+#include "util/minmax.h"
+#include "util/misc.h"
+
+typedef struct {
+  t_docId id;
+  t_expirationTimePoint point;
+} DocumentExpiration;
+
+typedef struct {
+  DocumentExpiration documentExpiration;
+  FieldExpiration* fieldExpirations;
+} TimeToLiveEntry;
+
+static uint64_t hashFunction_DocId(const void *key) {
+  return (uint64_t)key;
+}
+static void *dup_DocId(void *p, const void *key) {
+  return (void*)key;
+}
+static int compare_DocId(void *privdata, const void *key1, const void *key2) {
+  return (t_docId)key1 < (t_docId)key2;
+}
+static void destructor_DocId(void *privdata, void *key) {
+}
+
+static void destructor_TimeToLiveEntry(void *privdata, void *key) {
+    TimeToLiveEntry* entry = (TimeToLiveEntry*)key;
+    array_free(entry->fieldExpirations);
+    rm_free(entry);
+}
+
+static dictType dictTimeToLive = {
+  .hashFunction = hashFunction_DocId,
+  .keyDup = dup_DocId,
+  .valDup = NULL,
+  .keyCompare = compare_DocId,
+  .keyDestructor = destructor_DocId,
+  .valDestructor = destructor_TimeToLiveEntry,
+};
+
+void TimeToLiveTable_Init(TimeToLiveTable *table) {
```

Comment:
If we intend to use it every time we set a TTL, consider renaming to `VerifyInit`